src/main/scala/com/mongodb/spark/sql/fields/Field.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright 2016 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.spark.sql.fields
+
+import scala.util.Try
+
+protected trait Field[T] {
```

Comment:
In MongoDB parlance, "field" typically encompasses both the key and the value, as in "The projection document limits the fields to return for all matching documents.".  Given that, "FieldValue" or "Value" would be a more apt name.